grabbit/src/main/groovy/com/twcable/grabbit/ClientJobStatus.groovy

```diff
@@ -59,27 +82,45 @@ class ClientJobStatus {
         }
         new ClientJobStatus(
             jobExecution.id,
+            jobExecution.transactionID,
             jobExecution.startTime,
             jobExecution.endTime,
             jobExecution.exitStatus,
             jobExecution.jobParameters.getString(ClientBatchJob.PATH),
             timeTaken,
-            jobExecution.stepExecutions?.find { it.stepName == "clientJcrNodes" }?.writeCount ?: -1
+            (jobExecution.stepExecutions?.find { it.stepName == "clientJcrNodes" }?.writeCount ?: -1) as Integer
         )
     }
 
+    /**
+     * @param explorer The job explorer to query.
+     * @return Get the status of all jobs ever run. Will return an empty collection if no jobs have ever been run.
+     */
+    @Nonnull
+    static Collection<ClientJobStatus> getAll(@Nonnull JobExplorer explorer) {
+        if(explorer == null) throw new IllegalArgumentException("JobExplorer == null")
 
-    public static List<ClientJobStatus> getAll(@Nonnull JobExplorer explorer) {
-        if (explorer == null) throw new IllegalArgumentException("JobExplorer == null")
-
-        //Returns all the job instances that are currently running
+        //Returns all the job instances
         def instances = explorer.getJobInstances(ClientBatchJob.JOB_NAME, 0, Integer.MAX_VALUE)
         if (!instances) return Collections.EMPTY_LIST
 
-        final List<JobExecution> executions = instances.collectMany { explorer.getJobExecutions(it) }
-        final List<Long> jobExecutionIds = executions.collect { it.id }
+        final Collection<JobExecution> executions = instances.collectMany { explorer.getJobExecutions(it) }
+        final Collection<Long> jobExecutionIds = executions.collect { it.id }
         jobExecutionIds.collect { Long id ->
             get(explorer, id)
         }
     }
+
+    /**
+     * @param explorer The job explorer to query.
+     * @param transactionID to transactionID representing the transaction to query.
+     * @return a Collection of {@link ClientJobStatus} representing all jobs as part of a transaction, as denoted by transactionID; or null if transactionID provided is invalid.
```

Comment:
" null if transactionID provided is invalid"

The code as written would not return null, but an empty collection.

What does "invalid" mean? That it's not positive? That there never was a ID matching? That it's "unknown" because the transaction used to exist but has been pruned?